### PR TITLE
Fix #359 - Change pullpolicy conditionally based on image tag

### LINKED ISSUE #359: Omit imagePullPolicy from the deployment created by Sonata operator
### Describe the bug

The operator sets imagePullPoiicy to `Always`. This is against the default of openshift, which is ifNotPresent , and against
the k8s default which should be also ifNotPresent in case a tag or SHA is used.

### Expected behavior

if there's a tag which is not 'latest' , then omit the imagePullPolicy completely

### Actual behavior

imagePullPolicy is set in the generated deployment

### How to Reproduce?

1.Deploy any sonataflow resource, e.g greeting and see the default

oc create -f  https://raw.githubusercontent.com/rgolangh/serverless-workflows-helm/main/charts/workflows/charts/greeting/templates/01-sonataflow_greeting.yaml

2. oc get deployments greeting  -o jsonpath={.spec.template.spec.containers[]}  | jq '.imagePullPolicy'

### Output of `uname -a` or `ver`

k8s kind 0.20

### Golang version

_No response_

### Operator-sdk version

_No response_

### SonataFlow Operator version or git rev

_No response_

### Additional information
https://issues.redhat.com/browse/FLPATH-892
_No response_

### PR DESCRIPTION
**Pull request by ricardozanini**
**Description of the change:**
In this PR, we add the ability to pick the `PullPolicy` in the containers we deploy based on the image tag to follow Kubernetes/OpenShift conventions.

**Motivation for the change:**
Fix #359 

**Checklist**

- [x] Add or Modify a unit test for your change
- [x] Have you verified that tall the tests are passing?

<details>
<summary>
How to backport a pull request to a different branch?
</summary>

In order to automatically create a **backporting pull request** please add one or more labels having the following format `backport-<branch-name>`, where `<branch-name>` is the name of the branch where the pull request must be backported to (e.g., `backport-7.67.x` to backport the original PR to the `7.67.x` branch).

> **NOTE**: **backporting** is an action aiming to move a change (usually a commit) from a branch (usually the main one) to another one, which is generally referring to a still maintained release branch. Keeping it simple: it is about to move a specific change or a set of them from one branch to another.

Once the original pull request is successfully merged, the automated action will create one backporting pull request per each label (with the previous format) that has been added.

If something goes wrong, the author will be notified and at this point a manual backporting is needed.

> **NOTE**: this automated backporting is triggered whenever a pull request on `main` branch is labeled or closed, but both conditions must be satisfied to get the new PR created.
</details>

### CODE REVIEW

**Comment by ricardozanini** [comment]
I think there's a problem with ubuntu repos:

```
E: Failed to fetch http://security.ubuntu.com/ubuntu/pool/main/s/systemd/libudev-dev_249.11-0ubuntu3.11_amd64.deb  404  Not Found [IP: 52.252.163.49 80]
Fetched 541 kB in 1s (797 kB/s)
E: Unable to fetch some archives, maybe run apt-get update or try with --fix-missing?
Error: Process completed with exit code 100.
```

I'll check tomorrow.

**Review by domhanak — APPROVED** [review]

**Comment by ricardozanini** [comment]
I broke down the GHA fix in #365.

As soon as we merge that, I'll rebase here.

File: controllers/profiles/common/object_creators.go [hunk 1]
```diff
@@ -147,7 +147,6 @@ func defaultContainer(workflow *operatorapi.SonataFlow) (*corev1.Container, erro
 			FailureThreshold:    healthStartedFailureThreshold,
 			PeriodSeconds:       healthStartedPeriodSeconds,
 		},
-		ImagePullPolicy: corev1.PullAlways,
```
Comment (wmedvede) [inline controllers/profiles/common/object_creators.go:150]: When we are deploying workflows for managed builds, we don't use the "latest" tag, so this is good.
But when we deploy workflows with images that has been built outside SonataFlow we are not setting any ImagePullPolicy here. But, ¿should we analyze the image that was produced externally? or this produced at a later point?

**Review by wmedvede — COMMENTED** [review]
LGTM,silly question from my side only.

File: controllers/profiles/common/object_creators.go (hunk 1 above)
Comment (ricardozanini) [reply controllers/profiles/common/object_creators.go:150]: We are, the mutator that applies the image to the deployment and also verifies if an external image has the latest tag or not. So it depends on the user to set `latest` or an actual tag version to their image.

**Review by domhanak — APPROVED** [review]

File: controllers/profiles/common/object_creators.go (hunk 1 above)
Comment (wmedvede) [reply controllers/profiles/common/object_creators.go:150]: thanks!

**Review by wmedvede — APPROVED** [review]

**Pull request merged**